### PR TITLE
configurer: refactor startup secrets

### PR DESCRIPTION
**Pull request by wazery**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | mentioned in #1520, #1519, #1514, #1513, #1532 
| License         | Apache 2.0

### What's in this PR?
This PR is part of a series of feature PRs to implement purging unmanaged resources from Vault, you can refer to the previous PR over here #1532.

This PR just refactors StartupSecrets to be consistent with the new structure.

### Next

- Documentation and examples @aabouzaid will work on it.

### Checklist

- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)

### CODE REVIEW

File: internal/vault/startup_secrets.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+	"emperror.dev/errors"
+	"github.com/spf13/cast"
+)
+
+type startupSecret struct {
+	Type string `json:"type"`
+	Path string `json:"path"`
+	Data struct {
+		Data         map[string]interface{} `json:"data"`
```
Comment (aabouzaid) [inline internal/vault/startup_secrets.go:28]: I just realized that my code here is wrong, there is no data inside data but only a special case of `secretKeyRef`.

so anything under `startupSecret.Data` should be created as secret normally unless `secretKeyRef` is present there.

I will fix it tomorrow, please don't merge till that.

Comment (aabouzaid) [reply internal/vault/startup_secrets.go:28]: @bonifaido can you please confirm my observation?

so if we have this as in many examples:

```
startupSecrets:
  - type: kv
    path: secret/foo
    data:
      data:
        secret: foo
``

in the path `secret/foo` we will have a map called `{"data": {"secret": "foo"}} `.
if that is the case it's super confusing to have that as an example ... the second `data` key is actually part of the secret.

Comment (bonifaido) [reply internal/vault/startup_secrets.go:28]: If I remember correctly this used the lower level API (which has no idea about KV version 1 or 2), take this example, which would like to have the secrets in kv2 format, so you have to put manually the `/data/` section to the path and data wrapper to to the json itself:
```
    startupSecrets:
      - type: kv
        path: secret/data/accounts/aws
        data:
          data:
            AWS_ACCESS_KEY_ID: secretId
            AWS_SECRET_ACCESS_KEY: s3cr3t
```

Comment (aabouzaid) [reply internal/vault/startup_secrets.go:28]: @bonifaido thanks for the clarification, I've fixed it according to that.

**Comment by aabouzaid** [comment]
This PR is ready to be merged.

**Comment by aabouzaid** [comment]
@bonifaido done :heavy_check_mark:

**Review by bonifaido — APPROVED** [review]
LGTM! Thanks for this great work for all these PRs!

**Pull request merged**